### DISCUSSION TITLE
Adding middleware to internal route?

### DISCUSSION THREAD

**Discussion by leogomide** [pocketbase/pocketbase#5417] [Q&A] [answered]
Can i add a middleware to a internal route?

I create a 'posts' collection in the admin UI, is vae a field 'userId' that links to the table 'users', and i want to add the $apis.requireAdminOrOwnerAuth('userId') middleware to the 'posts' route.

I dont know if its possibel, and I was able to get a similar result with the hook below, but i think with the middlewares the code will be cleaner for the hooks...

```javascript
onRecordsListRequest((e) => {
    const requestInfo = $apis.requestInfo(e.httpContext);
        if (!requestInfo.authRecord && !requestInfo.admin) {
            throw new ApiError(403, 'onRecordsListRequest -> Unauthorized');
        }
}, 'posts')
```

**Comment by ganigeorgiev** [comment, marked as answer]
It is not clear what you are trying to do but you can't change the internal route (or at least not without hacky workarounds).

Note also that the above hook doesn't guarantee that `requestInfo.authRecord` is the owner of the post (assuming that this is what you are trying to do).

Additionally for the records CRUD operations I recommend to rely on the collection API rules because the rules are also used for example as part of other APIs like the realtime subscriptions, expand, etc.

An example List API rule that will allow only "owners" (and admins, since admins always have access to everything) could be:

```
// assuming that userId is a field in the "posts" collection
@request.auth.id != "" && @request.auth.id = userId
```

For more details please refer to https://pocketbase.io/docs/api-rules-and-filters/.

**Reply by leogomide** [reply]
ok, got it

the behaviour i want to prevent, is the api answer result 200 when a invalid or missin key is sended

if i apply this rule like you say, if i send a request with invalid key, it will return 200 with a blank list and zero count, i want it to return 403, unauthrized

as said, the code above give me the bahaviour i want, i will apply together with the filter rules, thanks for your time 🤝
🙏
